utils/preprocessing.py

```diff
@@ -147,38 +169,79 @@ def sample2numpy(sample: dict, bands_to_delete: List[str], w: int=25, h: int=25,
         for r_, row in enumerate(band):
-            array[r_, :, b_] = row
-    return array
+            array[b_, :, r_] = row
+    return array.astype(np.float32)
+
+
+
+def data_augmentation(array: np.array, transforms: List[partial], output_path: str):
+    '''
+    This function performs data augmentation on the given array based on a list of transforms. For
+    each given transform, a new npy file is written with the naming convention:
+    "<output_path>-<name_of_transform>.npy"
+
+    array: array that should be augmented
+    transforms: List of transformer functions from torchvision.transforms.v2.functional
+                (https://pytorch.org/vision/stable/transforms.html#transform-classes-functionals-and-kernels)
+                that are used for data augmentation of the arrays
+    output_path: output path of array without ".npy" at the end
+    '''
+    transforms = list(set(transforms)) # only use unique transforms
+    # loop over all transforms
+    for transform in transforms:
+        transf_array = transform(torch.from_numpy(array)).numpy() # apply transform
+        transform_name = transform.func.__name__ # get name of function
+        # add name and value of function keywords
+        if len(transform.keywords):
+            for key, value in zip(transform.keywords.keys(),transform.keywords.values()):
+                transform_name = transform_name + '-' + key + '=' + str(value)
+        # save numpy file
+        np.save(f'{output_path}-{transform_name}.npy', transf_array, allow_pickle=False)
+
 
 
 ##############################
 
 
 if __name__ == "__main__":
-    identifier = 1102
+    identifier = 1123
     data_dir = 'data'
 
     bands_to_delete = ["B2"]
 
-    # Preproessing options ################
+    # PREPROCESSING OPTIONS ##################################################################################
     #what_happens_to_nan='apply_nan_mask'
     what_happens_to_nan='delete_nan_samples'
     #what_happens_to_nan='keep_nan'
 
-    # Transformer options ################
-    #transformer = None
-    transformer = partial(np.nanmean, axis=(0,1))  # mean per band excluding nan values
-    #transformer = partial(np.nanmedian, axis=(0,1)) # median per band excluding nan values
+    # TRANSFORMER ############################################################################################
+    # either use transformer_for_numpy_array or transformers_data_augmentation
+
+    # transformer_for_numpy_array
+    transformer = None
+    #transformer = partial(np.nanmean, axis=(1,2))  # mean per band excluding nan values
+    #transformer = partial(np.nanmedian, axis=(1,2)) # median per band excluding nan values
+
+    # transformers_data_augmentation
+    # see https://pytorch.org/vision/stable/transforms.html#transform-classes-functionals-and-kernels
+    # only use functional functions!
+    from torchvision.transforms.v2 import functional
+    data_aug_transformers = [partial(functional.horizontal_flip),
```

Comment:
wrapping this in a partial seems a bit overkill here, just the object would do just fine I think

Reply:
Ah, but above extraction of keywords would probably not work right?

Reply:
No, rather

```python
data_aug_transformers = [functional.horizontal_flip,
                         functional.vertical_flip,
                         partial(functional.rotate, angle=180)]
```

But this would require more changes to the info extraction, something like this

```python
for transform in transforms:
    transf_array = transform(torch.from_numpy(array)).numpy() # apply transform
    
    if isinstance(transform, partial):
        transform_name = transform.func.__name__ # get name of function
        # add name and value of function keywords
        if len(transform.keywords):
            for key, value in zip(transform.keywords.keys(),transform.keywords.values()):
                transform_name += + '-' + key + '=' + str(value)
    else:
        transform_name = transform.__name__
```